ddl/constraint.go

```diff
@@ -77,3 +329,51 @@ func findDependedColsMapInExpr(expr ast.ExprNode) map[string]struct{} {
 	}
 	return colsMap
 }
+
+func (w *worker) addTableCheckConstraint(dbInfo *model.DBInfo, tableInfo *model.TableInfo, constr *model.ConstraintInfo, job *model.Job) error {
+	// Get sessionctx from ddl context resource pool in ddl worker.
+	var ctx sessionctx.Context
+	ctx, err := w.sessPool.get()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer w.sessPool.put(ctx)
+
+	// If there is any row can't pass the check expression, the add constraint action will error.
+	// It's no need to construct expression node out and pull the chunk rows through it. Here we
+	// can let the check expression restored string as the filter in where clause directly.
+	// Prepare internal SQL to fetch data from physical table under .
+	sql := fmt.Sprintf("select * from `%s`.`%s` where ", dbInfo.Name.L, tableInfo.Name.L)
+	sql = sql + " not " + constr.ExprString + " limit 1"
```

Comment:
Why not put it all in `Sprintf`?

Reply:
Why use `select *` rather than `select count(1)`?

Reply:
make sense